docs/decisions/backend/0009-advanced-installer-design.md

```diff
@@ -0,0 +1,246 @@
+  to provide appropriate hints (suggestions) for them to deploy an unknown mod; even in presence of install
+  instructions from mod authors.
+
+- Must support arbitrary path mapping.
+
+  This system must support mapping any path from the source archive into any location in the target game directorie(s).
+
+  This includes mapping multiple files in the same folder to different locations, in case the modder provides
+  multiple files side by side when you are only supposed to pick 1.
+
+- Must recognise conflicts.
+
+  The end user using the advanced installer, might end up trying to map files in a way that could cause a conflict.
+
+  For example, the user might map a file to the output directory individually, then map an entire folder; and a file
+  within the folder conflicts with an already mapped file.
+
+## An Example of Manual Installation
+
+Some mods may have multiple variants, but don't use a declarative language like the FOMOD XML
+format for specifying choices. This tends to be more common in games with more niche communities
+and early mods for games.
+
+Imagine the following file structure (imaginary game):
+
+```text
+Red Hair
+├─ Data
+│  ├─ model.dae
+│  └─ hair.dds
+Green Hair
+├─ Data
+│  ├─ model.dae
+│  └─ hair.dds
+Blue Hair
+└─ Data
+   ├─ model.dae
+   └─ hair.dds
+```
+
+The user is expected to select items from one of the given folders. These folders do map to game folder, but
+because there are duplicates and multiple candidates for deployment, the Nexus App cannot automatically
+choose which one to deploy.
+
+## Considered Options (Deployment)
+
+- Reusing the FOMOD Deployment System.
+- Programming a new Deployment System from scratch.
+
+## Decision Outcome
+
+After some investigation work, I it has been concluded that it isn't really possible to reuse the FOMOD system here.
+
+While the FOMOD system is technically a subset of the Advanced Installer if you think about it (after all, toggling
+1 FOMOD option is just like mapping 1-* files to a given location); the design of the current FOMOD code does not
+easily lend itself to reuse.
+
+This is because FOMOD library, emits a list of files to be deployed from the library as 'Instructions'; and we simply
+use those instructions to modify an enumerable of files to be deployed. This file list is created by user interacting with
+the FOMOD library in an asynchronous call (i.e. logic abstracted away by library). As a result, reusing existing FOMOD
+logic does not translate well to `Advanced Installer`,
+
+On the other hand, a good chunk of the UI styling/logic should be reusable.
+In other words, we're going for a custom solution.
+
+### Consequences
+
+- Good, because custom solution allows for easier serialization of the data.
+- Good for maintenance, as the logic for persisting deployment steps could be reused in the FOMOD code.
+    - Note: Both are a file->file mapping effectively, so we can use serialization format for Advanced Installer in FOMOD.
```

Comment:
The deployment steps are already shared between all mods, mod installation has no baring on it. Installer just returns a list of source/target mappings and everything else is handled by common code.